### PR TITLE
[FEATURE] Nested workspaces with access restrictions

### PR DESCRIPTION
**Pull request by robertlemke**
This is a major feature for Neos which allows to create and nest workspaces, assign ownership to workspaces and thus restrict a group of users to access / modify certain workspaces.

### CODE REVIEW

**Comment by robertlemke** [comment]
This WIP is basically working, however there is still some work do be done to get this change polished, most importantly adjustments to the security policy (probably with a new, custom Workspace Privilege type) and tests.

If you're interested in this feature, feel free to try it out already and give me feedback here.

File: TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/Workspace.php [hunk 1]
```diff
@@ -180,8 +193,65 @@ public function setDescription($description)
...
+
+    /**
+     * Checks if this workspace is shared accross all editors
+     *
+     * @return boolean
+     * @api
+     */
+    public function isInternalWorkspace()
+    {
+        return $this->owner === null;
```
Comment (robertlemke) [inline TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/Workspace.php]: should probably check if it's not a personal workspace

Comment (kdambekalns) [inline TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/Workspace.php]: Fixed with 3427326

**Comment by robertlemke** [comment]
Pushed a change which decouples TYPO3CR from Neos again. The MySQL migrations work fine (they initially set personal workspace owners to the respective Neos users) but we still need to provide a PostgreSQL migration.

**Comment by robertlemke** [comment]
We are getting closer to the final version for this PR. What's still on our todo:

- verify security policies for the AJAX service (policies are there, but should be cross-checked)
- documentation containing basic terminology and general concept

**Comment by robertlemke** [comment]
Some additional background information: This PR also introduces a new "owner" field to the `Workspace` model in TYPO3CR. The owner of a workspace is a Neos user. But in order to avoid a dependency from TYPO3CR to Neos, I simply store a UUID and introduced a new `UserInterface` in TYPO3CR. I also experimented with using Doctrine's (ResolveTargetEntityListener)[http://doctrine-orm.readthedocs.org/projects/doctrine-orm/en/latest/cookbook/resolve-target-entity-listener.html] but that would require additional support for interface as entities in Flow, which would be too much work for the purpose right now. Take a look at the `Workspace` class to check how I solved it - I think it's okay that way, pragmatic, but also not really dirty.

**Comment by robertlemke** [comment]
Great, thank you @kdambekalns for the documentation!

**Comment by kdambekalns** [comment]
Just tested the changes done in the last commit, works fine.

**Comment by kdambekalns** [comment]
Reviewed, tested, :+1: from my side

**Comment by robertlemke** [comment]
:+1:

**Comment by robertlemke** [comment]
:dancers: :fireworks: :

**Pull request merged**